packages/filesystem/src/browser/filesystem-frontend-contribution.ts

```diff
@@ -124,13 +124,14 @@ export class FileSystemFrontendContribution implements FrontendApplicationContri
         return !environment.electron.is() && fileStat.isDirectory;
     }
 
-    protected async upload(selection: FileSelection): Promise<void> {
+    protected async upload(selection: FileSelection): Promise<FileUploadResult | undefined> {
         try {
             const source = TreeWidgetSelection.getSource(this.selectionService.selection);
-            await this.uploadService.upload(selection.fileStat.resource);
+            const fileUploadResult: FileUploadResult = await this.uploadService.upload(selection.fileStat.resource);
```

Comment:
nit: `: FileUploadResult` is redundant, it will be inferred from `upload`

It is a good practice to have explicit return types to allow typescript to infer types for variables, i.e. redeclaring them in such case is always redundant.